src/Writers/MarkdownWriter.jl

```diff
@@ -145,7 +145,9 @@ function render(io::IO, mime::MIME"text/plain", d::Documents.MultiOutput, page,
 end
 function render(io::IO, mime::MIME"text/plain", d::Dict{MIME,Any}, page, doc)
     filename = String(rand('a':'z', 7))
-    if haskey(d, MIME"text/html"())
+    if haskey(d, MIME"text/markdown"())
+        println(io, d[MIME"text/markdown"()])
+    elseif haskey(d, MIME"text/html"())
```

Comment:
I wonder if we should actually prefer HTML output here since it would in general be richer than Markdown?

Reply:
Since users can put their object in a wrapper object that is showable only as markdown or html (like `Base.Text` does), I think this is not a super big issue.  I'm in favor of trying to print text/markdown first because:

* Markdown files should be human-readable in principle.

    * Counter argument: We are using markdown as "intermediate representation" so it doesn't have to be human-readable.

* Rules for inlining html is non-trivial: https://spec.commonmark.org/0.28/#html-blocks (OK, this is commonmark, not markdown, but I suspect that commonmark compliant parser would be the main consumer of the MarkdownWriter output).  So, it would break markdown parser if `show` prints arbitrary html.

    * Counter argument: It is probably possible to post-process html so that it is treated as a single html block by commonmark parsers.

* It let us use some markdown extensions for downstream renderer (pandoc, gitlab wiki, etc.).

    * Counter argument: As I said above, it's not hard to choose preferred mime.

Since I can find counter argument to my reasoning, I don't really know which is better.  But I still prefer trying markdown first :)

Reply:
My assumption in general is that it will be consumed by MkDocs, which uses [Python-Markdown](https://python-markdown.github.io/), which in turn implements [the original John Gruber Markdown](https://daringfireball.net/projects/markdown/).

However, I think you are right that using Markdown would be more reliable, especially as we have no control over how objects decide to encode themselves into HTML. And, as you say, if need be, this can be overriden by the user. So let's keep it as it is :slightly_smiling_face: